compiler/rustc_arena/src/lib.rs

```diff
@@ -74,7 +74,7 @@ impl<T> ArenaChunk<T> {
     #[inline]
     unsafe fn new(capacity: usize) -> ArenaChunk<T> {
         ArenaChunk {
-            storage: NonNull::new(Box::into_raw(Box::new_uninit_slice(capacity))).unwrap(),
+            storage: NonNull::new_unchecked(Box::into_raw(Box::new_uninit_slice(capacity))),
```

Comment:
This is a very nice example of why `unsafe_op_in_unsafe_fn` not being on by default is a big mistake. The safety condition of the function (which I actually don't know because it's not documented :/) has nothing to do with the safety condition of `new_umchecked`.

I'd love to take a PR denying that lint on rustc_arena if you want to.

Reply:
Isn't this better as `NonNull::from(Box::leak(...))`?